config.py

```diff
@@ -208,6 +208,16 @@ class DefaultConfig(ImmutableConfig):
 
     @staticmethod
     def create_transaction(db):
+        # This hack is for handling possible MySQL closing idle connections.
+        # Peewee/pymysql's will try to reuse the existing connection after the MySQL server
+        # has already closed it, which will return an InterfaceError.
+        # AFAIK, there isn't a way to actually test for a a stale connection without actually
+        # running a query. Closing the connection before the transaction forces peewee/pymysql
+        # to reopen a new session to MySQL. This should only applies to non-registry workers,
+        # as the registry workers use a pool by default, and shouldn't have this issue.
+        if type(db.obj).__name__ == "ObservableRetryingMySQLDatabase":
```

Comment:
is this problem specific to Mysql? Where is the string `ObservableRetryingMySQLDatabase` coming from? Can we use `isinstance` instead ?

Reply:
We're using the class name as we can't easily check the for the instance type as they're defined dynamically at runtime:
- https://github.com/quay/quay/blob/master/data/database.py#L380-L381
- https://github.com/quay/quay/blob/master/data/database.py#L465-L468

From what I can tell, this is is a problem specific how MySQL cleans up idle connection based on `wait_timeout`, and peewee/PyMySQL seemingly unable to handle it gracefully.